### PR TITLE
Fix for XSS issues on JobCalendar VF page

### PR DESCRIPTION
**Pull request by seanpat09**
# Critical Changes

# Changes
Security Issues addressed on JobCalendar page.

# Issues Closed
Fixes XSS issues on JobCalendar due to custom labels used in JavaScript

# New Metadata

# Deleted Metadata

### CODE REVIEW

**Comment by seanpat09** [comment]
**lurch: attach W-026537

**Comment by LurchTheButler** [comment]
Tracking <a href="https://foundation.my.salesforce.com/a2x8000000AYCDxAAP" target="_blank">W-026537</a>

**Comment by jjbennett** [comment]
In NPSP we are putting "Security Issues addressed on {!pageName} page. In the changes section of the Release Notes, can you do that here as well. Side note, starting to review now. Thanks!

**Comment by seanpat09** [comment]
@jjbennett thanks for the feedback, I'll make those changes.

As for this solution in general, I agree with your concerns. What I like about this solution is that it works with the current format of the labels. If the customer can update or add more translations themselves, then this will assure that we don't break their page. If they can't, then we should consider changing the format of the labels or even using different labels.

If we want to continue using key/value pairs like this, then we should just make the labels into valid JSON strings. That way we can just do something like `JSON.parse('{!$Label.label calendarButtonText}')` and bypass the need for `UTIL_JavaScriptSanitizer.cls` completely.

I think for the sake of fixing this security hole I think you're right in that we mark this as a workaround for now. This page is a couple years old now so it would be worth it to figure out what's the best way to allow user customization of a page using newer salesforce features.

File: src/classes/UTIL_JavaScriptSanitizer.cls [hunk 1]
```diff
@@ -96,10 +96,8 @@ public with sharing class UTIL_JavaScriptSanitizer {
     private String replaceCommasWithSplitMarkers(String toParse) {
         Boolean inQuotes = false;
         String openQuoteChar;
-        Integer i = 0;
-
         String substitutedValue = '';
-        do {
+        for ( Integer i = 0; i < toParse.length(); i++ ){
```
Comment (jjbennett) [inline src/classes/UTIL_JavaScriptSanitizer.cls]: Should this make that if statement no longer necessary?

Comment (jjbennett) [reply src/classes/UTIL_JavaScriptSanitizer.cls]: P.S. Extra spaces after the open ( and before the close and missing after the open :)

Comment (seanpat09) [reply src/classes/UTIL_JavaScriptSanitizer.cls]: ah good catch! It certainly is Friday

**Comment by seanpat09** [comment]
@rponti-sforg This is ready for testing!

**Comment by seanpat09** [comment]
Fixes #376

**Pull request merged**